spec/lib/search_spec.rb

```diff
@@ -94,4 +94,21 @@
       end
     end
   end
+
+  context "custom_eager_load" do
+    it "includes custom tables" do
+      expect(Search.new("test").send(:posts_eager_loads, Post).includes_values).to eq([:user, :post_search_data, { topic: [:category] }])
+
+      SiteSetting.tagging_enabled = true
+      expect(Search.new("test").send(:posts_eager_loads, Post).includes_values).to eq([:user, :post_search_data, { topic: [:category, :tags] }])
+
+      Search.custom_topic_eager_load([:test_array])
```

Comment:
Since `Search.custom_topic_eager_load` uses a class variable, we need to clean this up to prevent this test from leaking state.